flink-runtime/src/test/java/org/apache/flink/runtime/operators/sort/ExternalSortLargeRecordsITCase.java

```diff
@@ -71,7 +71,7 @@ public void beforeTest() {
 	}
 
 	@After
-	public void afterTest() {
+	public void afterTest() throws Exception {
 		this.ioManager.close();
 		if (!this.ioManager.isProperlyShutDown()) {
```

Comment:
looking at this it makes it seem like ioManager.close should never throw exceptions. We can already see downsides of this change since, if close() throws an exception, 
a) isProperlyShutDown is never called
b) the memory manager would not be shutdown

Reply:
we could add try catch and ignore the exception here. WDYT?

Reply:
it's not just this test though, there are plenty of others having the same problem.

I'm wondering what value we're adding by changing this behavior. For production at least it makes no difference since we still just log the exception.

Reply:
Yes, it might involve in many tests.

Actually my first version was ignoring any exceptions thrown by `IOManager#close`, and @azagrebin suggested ignoring only `IOException` and throwing any unexpected exceptions outside to make quickly fail. 

Actually I think these two options both work on my side. If we revert to previous behavior, we did not change anything during refactoring, otherwise we changed the behavior as now to do as we think in more proper way.

Reply:
most of the test were failing anyway if the ioManager shutdown failed due to subsequent assertions on `isProperlyShutdown`; thus b) is no longer a new problem.

I would suggest to go through all tests, remove the `isProperlyShutDown` call if the test follows the same pattern, and potentially remove the method altogether if it is unused at that point.

Reply:
Agree, that is a good finding. 
I could solve this previous existing issue in a separate hotfix commit as precondition.

Reply:
After thinking through it further, we might not remove the `isProperlyShutDown` call directly, because it is used for checking previous `IOException` which was swallowed during `close`. And `close` itself could only thrown non `IOException` atm.

I also do not like the way of `isProperlyShutDown`, if we want to remove it completely, another option is let `IOManager#close` also throw `IOException` after all the file delete loop. WDYT?

Reply:
After confirming with andrey, we could keep the previous behavior to still ignore `IOException` during close, because we just try best to delete files for releasing resources. If one file was already deleted by other factors before, it is no need to cause unnecessary failover. 

If we want to refactor this behavior future, we could do it separately, not to block this PR. I remove `try catch` from `IOManager#close` and add some javadoc for this method.